hq/app/logic/CredentialsReportDisplay.scala

```diff
@@ -104,24 +99,20 @@ object CredentialsReportDisplay {
   }
 
   def reportStatusSummary(report: CredentialReportDisplay): ReportSummary = {
-    val reportStatusSummary = report.humanUsers.map(_.reportStatus) ++ report.machineUsers.map(_.reportStatus)
+    val reportStatusSummary = (report.humanUsers ++ report.machineUsers).map(_.reportStatus)
 
-    val (warnings, errors, other) = reportStatusSummary.foldLeft(0,0,0) {
-      case ( (war, err, oth), Amber ) => (war+1, err, oth)
-      case ( (war, err, oth), Red ) => (war, err+1, oth)
-      case ( (war, err, oth), Blue ) => (war, err, oth+1)
-      case ( (war, err, oth), _ ) => (war, err, oth)
-    }
-    ReportSummary(warnings, errors, other)
+    val warnings = reportStatusSummary.count({ case Amber => true })
+    val errors = reportStatusSummary.count({ case Red => true })
+    val others = reportStatusSummary.count({ case Blue => true })
+
+    ReportSummary(warnings, errors, others)
   }
 
   def exposedKeysSummary(allReports: Map[AwsAccount, Either[FailedAttempt, List[ExposedIAMKeyDetail]]]): Map[AwsAccount, Boolean] = {
-    for {
-      (account, report) <- allReports
-    } yield (account, report match {
-        case Right(keys) if keys.nonEmpty => true
-        case _ => false
-    })
```

Comment:
here we're doing an operation on the values of a map only - so `mapValues` is better suited